### PR TITLE
Closes #170 - added discretized restoration output to water facility restoration

### LINKED ISSUE #170: water facility restoration should output discretized functionality based on damage
Water facility restoration should output discretized functionality based on damage. This will require the following changes:

1. The inventory dataset input should be replaced by the damage input
2. New output that gives discretized restoration for day 1, 3, 7, etc

The original analysis output will still be available for those who want to combine that output with other approaches to functionality.

### PR DESCRIPTION
**Pull request by navarroc**
This follows the same logic as electric power facility restoration. Although the hazus table layout is the same, I created a new type for the discretized restoration. It could be argued that we should use the same type and ingest the tables using the same data type. EPF came from Table 8-28, the water facility came from Table 8-2. There are two new inputs for Water facility restoration:

1. Water facility damage to provide the damage probabilities
2. A new input dataset: incore:waterFacilityDiscretizedRestoration from Table 8-2 (ingested as 62cc7e62861e370172c7e2f1)

The unit test is updated to chain water facility damage and use the new input referenced above.

### CODE REVIEW

File: pyincore/analyses/waterfacilityrestoration/waterfacilityrestoration.py [hunk 1]
```diff
@@ -48,33 +48,40 @@ def run(self):
...
-                                                                                               pf_interval)
+        restoration_table_dataset = self.get_input_dataset("restoration_table").get_csv_reader()
+        restoration_table = AnalysisUtil.get_csv_table_rows(restoration_table_dataset, ignore_first_row=False)
+
+        damage = self.get_input_dataset("damage").get_csv_reader()
+        damage_result = AnalysisUtil.get_csv_table_rows(damage, ignore_first_row=False)
+
+        (inventory_restoration_map, pf_results, time_results, func_results) = self.waterfacility_restoration(
+            inventory_list, damage_result, mapping_set, restoration_key, end_time, time_interval, pf_interval,
+            restoration_table)
```
Comment (nunezco2) [inline pyincore/analyses/waterfacilityrestoration/waterfacilityrestoration.py:60]: Is the format aligned with PEP8?

**Review by nunezco2 — APPROVED** [review]

**Comment by ywkim312** [comment]
can you add the paths to the files in test_format.py so the formats of the files be tested?

**Comment by ywkim312** [comment]
test ran successfully. I will approve when test_format.py file has the path for changed files and test_format get passed.

File: pyincore/analyses/waterfacilityrestoration/waterfacilityrestoration.py (hunk 1 above)
Comment (navarroc) [reply pyincore/analyses/waterfacilityrestoration/waterfacilityrestoration.py:60]: I don't see any pep8 warnings and reformat code doesn't change anything about that line.

**Comment by navarroc** [comment]
> test ran successfully. I will approve when test_format.py file has the path for changed files and test_format get passed.

Done. Added all water facility files to test_format.py

**Comment by ywkim312** [comment]
> 
there's conflict in test_format.py, could you resolve it?

**Comment by navarroc** [comment]
> > 
> 
> there's conflict in test_format.py, could you resolve it?

done

File: tests/test_format.py [hunk 2]
```diff
@@ -62,7 +62,10 @@
     os.path.join(PYINCORE_ROOT_FOLDER, 'pyincore/analyses/housingrecovery/housingrecovery.py'),
     os.path.join(PYINCORE_ROOT_FOLDER, 'pyincore/analyses/housingrecovery/housingrecoveryutil.py'),
     os.path.join(PYINCORE_ROOT_FOLDER, 'tests/pyincore/analyses/housingrecovery/test_housingrecovery.py'),
-
+    os.path.join(PYINCORE_ROOT_FOLDER, 'pyincore/analyses/waterfacilityrestoration/waterfacilityrestoration.py'),
+    os.path.join(PYINCORE_ROOT_FOLDER, 'pyincore/analyses/waterfacilityrestoration/waterfacilityrestorationutil.py'),
+    os.path.join(PYINCORE_ROOT_FOLDER,
+                 'tests/pyincore/analyses/waterfacilityrestoration/test_waterfacilityrestoration.py')
```
Comment (ywkim312) [inline tests/test_format.py:68]: please add comma at the end of the line

Comment (navarroc) [reply tests/test_format.py:68]: Develop did not have a trailing comma so I followed that pattern. @nunezco2 what is your preference here? Trailing commas are subjective.

Comment (ywkim312) [reply tests/test_format.py:68]: It has to be separated with the next line by the comma

Comment (navarroc) [reply tests/test_format.py]: My bad, thought you meant adding a trailing comma at the end of the list. Fixed.

**Review by ywkim312 — APPROVED** [review]

**Pull request merged**